src/riscv/lib/src/pvm/linux/signals.rs

```diff
@@ -16,20 +18,133 @@ use crate::state::NewState;
 use crate::state_backend::AllocatedOf;
 use crate::state_backend::Atom;
 use crate::state_backend::Cell;
+use crate::state_backend::Elem;
 use crate::state_backend::FnManager;
 use crate::state_backend::ManagerAlloc;
 use crate::state_backend::ManagerBase;
 use crate::state_backend::ManagerClone;
+use crate::state_backend::ManagerRead;
 use crate::state_backend::ManagerReadWrite;
 use crate::state_backend::Ref;
 use crate::struct_layout;
 
+#[repr(C)]
+#[derive(Clone, Debug)]
+/// Linux sigaction struct, see <https://man7.org/linux/man-pages/man2/sigaction.2.html>
+pub struct LinuxSigAction {
+    sa_handler: VirtAddr,
+    sa_sigaction: VirtAddr,
+    sa_mask: u32,
+    sa_flags: u32,
+    sa_restorer: VirtAddr,
+}
+
+#[cfg(test)]
+// Currently we only support the `sa_sigaction` field, so in tests it's useful to be able to
+// create a sigaction using only this field.
+impl LinuxSigAction {
+    pub(crate) fn new(sa_sigaction: VirtAddr) -> Self {
+        Self {
+            sa_handler: VirtAddr::new(0),
+            sa_sigaction,
+            sa_mask: 0,
+            sa_flags: 0,
+            sa_restorer: VirtAddr::new(0),
```

Comment:
```suggestion
            sa_sigaction,
            ..Default::default()
```

(would require `#[cfg_attr(test, derive(Default))]` for `VirtAddr`)
```